### PR TITLE
Cleanmedianet: Add cleanmedianet bidder adapter

### PR DESCRIPTION
**Pull request by sa1omon**
## Type of change

[X] New bidder adapter

## Description of change

Added `cleanmedianet` as a new bidder adapter

```
var adUnits = [
    // Banner adUnit
   {
     code: 'banner-div',
     sizes: [[300, 250]],
     bids: [{
       bidder: 'cleanmedianet',
       params: {
          // ID of the supply partner you created in the Clean Media Net dashboard
         supplyPartnerId: '1253',
          // OPTIONAL: custom bid floor
         bidfloor: 0.01,
          // OPTIONAL: if you know the ad position on the page, specify it here
         //           (this corresponds to "Ad Position" in OpenRTB 2.3, section 5.4)
         //adpos: 1,
          // OPTIONAL: whether this is an interstitial placement (0 or 1)
         //           (see "instl" property in "Imp" object in the OpenRTB 2.3, section 3.2.2)
         //instl: 0
       }
     }]
   },
    // Video outstream adUnit
   {
     code: 'video-outstream',
     sizes: [[300, 250]],
     mediaTypes: {
       video: {
         context: 'outstream',
         playerSize: [300, 250]
       }
     },
     bids: [ {
       bidder: 'cleanmedianet',
       params: {
          // ID of the supply partner you created in the dashboard
         supplyPartnerId: '1254',
          // OPTIONAL: custom bid floor
         bidfloor: 0.01,
          // OPTIONAL: if you know the ad position on the page, specify it here
         //           (this corresponds to "Ad Position" in OpenRTB 2.3, section 5.4)
         //adpos: 1,
          // OPTIONAL: whether this is an interstitial placement (0 or 1)
         //           (see "instl" property in "Imp" object in the OpenRTB 2.3, section 3.2.2)
         //instl: 0
       }
     }]
   }
];
```

### CODE REVIEW

File: modules/cleanmedianetBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,307 @@
+import * as utils from '../src/utils';
+import { registerBidder } from '../src/adapters/bidderFactory';
+import { config } from '../src/config';
+import { Renderer } from '../src/Renderer';
+import { BANNER, VIDEO } from '../src/mediaTypes';
+
+export const helper = {
+  getTopFrame: function() {
```
Comment (mike-chowla) [inline modules/cleanmedianetBidAdapter.js]: Please use bidderRequest.refererInfo for getting the url
http://prebid.org/dev-docs/bidder-adaptor.html#bidrequest-parameters

File: modules/cleanmedianetBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    try {
+      return window.top === window ? 1 : 0;
+    } catch (e) {}
+    return 0;
+  },
+  startsWith: function(str, search) {
+    return str.substr(0, search.length) === search;
+  },
+  getTopWindowDomain: function(url) {
```
Comment (mike-chowla) [inline modules/cleanmedianetBidAdapter.js]: Please use utils.parse for extracting the hostname from the url
https://github.com/prebid/Prebid.js/blob/68dfb641a2ee27d9b538910ed92f253a4f6e3685/src/url.js#L27

File: modules/cleanmedianetBidAdapter.js (hunk 1 above)
Comment (sa1omon) [reply modules/cleanmedianetBidAdapter.js]: Fixed

File: modules/cleanmedianetBidAdapter.js (hunk 2 above)
Comment (sa1omon) [reply modules/cleanmedianetBidAdapter.js]: Fixed

**Comment by sa1omon** [comment]
> Please update to use the passed in url & utils url parsing function

Hi @mike-chowla,
I have updated the code according to your insights.
Please update me if there are any problems.
Thanks

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/1134

**Pull request merged**